src/v/kafka/server/handlers/fetch.h

```diff
@@ -68,6 +68,14 @@ struct op_context {
         }
         intrusive_list_hook _hook;
 
+        size_t response_size() {
+            auto& records_opt = _it->partition_response->records;
+            if (!records_opt) {
+                return 0;
+            }
+            return records_opt->size_bytes();
```

Comment:
nitpick:
```suggestion
            return _it->partition_response->records
              .transform(&kafka::batch_reader::size_bytes)
              .value_or(0);
```

Reply:
Oh, nice call. Switched to your suggestion.